kafka-rest/src/main/java/io/confluent/kafkarest/KafkaRestConfig.java

```diff
@@ -718,11 +756,17 @@ public Properties getProducerProperties() {
     addTelemetryReporterProperties(producerProps);
     producerProps.putAll(originalsWithPrefix("client.", /* strip= */ true));
     producerProps.putAll(originalsWithPrefix("producer.", /* strip= */ true));
-    producerProps.putAll(originalsWithPrefix("schema.registry", /* strip= */ false));
+    producerProps.putAll(getSchemaRegistryConfigs());
```

Comment:
why does the producer need SR configs? I thought you were changing things to do serialization / SR interaction before producing.

Reply:
That's because of the very restrictive interface in `Serializer`. The `serialize` method only takes `topicName` and the object to serialize. In order to serialize a schema-ed record, you need more information. Namely: `subject`, `schemaId`, and for protobuf whether you are serializing a key or a value. The serializer needs to acquire all this information out of band, since it can't be passed to `serialize`. So it will use schema registry for that. `subject` will be inferred using the `subject.name.strategy` schema registry config., while the `schemaId` will be acquired by calling schema registry. And the `isKey` thing will come from `Serializer.configure(..., isKey)`.

I have re-implemented `RecordSerializer` now to reuse the `SchemaRegistryClient` registed in `SchemaRegistryModule`. And I am now using `AbstractKafkaSchemaSerDe`, which exposes a method `serializeImpl` that allows to passing the subject, schema and isKey directly. So now the only call to schema registry from the serializer should be to acquire `schemaId`, and that call will likely return from the schema registry cache, since we just did the same in SchemaManager.

Unfortunately there's no method that accepts `schemaId` directly. Otherwise we would have removed all calls to `SchemaRegistryClient` from serializer. Anyways, I hope that provides enough context.

Reply:
Ow, yeah, now I understand what you mean. Yes, the producer does not need them. The problem is that I'm the producer configs to configure the serializers (which is what the producer does). Maybe I can separate them. Let me take a look :)

Reply:
Done.